web-portal/pages/submit-event.vue

```diff
@@ -101,132 +101,135 @@
 
 </template>
 
-<script>
+<script setup>
+  import { ref, onMounted, onBeforeUnmount, nextTick } from 'vue'
   import { useStore } from 'vuex'
+  import { useRoute, onBeforeRouteLeave } from 'vue-router'
   import SubmissionForm from '@/components/SubmissionForm.vue'
   import SubmissionPreview from '@/components/SubmissionPreview'
   import PartnerService from '@/services/PartnerService'
   import { FETCH_ACTIVE_VENUES } from '../store/venues'
 
-  export default {
-    async setup () {
-      definePageMeta({
-        layout: 'no-mobile-cta',
-      })
-      const { query } = useRoute()
-      const store = useStore()
-
-      // scaffold event (this is a synchronous operation)
-      store.dispatch('CreateNewEvent')
-      // fetch venues to power venue selector
-      await useLegacyStoreFetch(FETCH_ACTIVE_VENUES, [
-        FETCH_ACTIVE_VENUES
-      ])
-
-      // display partner info if provided in query
-      if (query.partner) {
-        return {
-          partner: PartnerService.getPartnerForQuery(query.partner)
-        }
-      }
-    },
-    data: function () {
-      return {
-        mode: 'edit',
-        previewEvent: null,
-        partner: null,
-        submissionErrorMessage: null,
-      }
-    },
-    // sometimes users navigate away without submitting
-    // if the form is partially filled, we can try to warn them and give them
-    // a chance to abort the nav
-    // mounted/beforeDestroy handle the "close tab / window" case...
-    mounted: function () {
-      if (window) {
-        window.addEventListener('beforeunload', this.promptIfInProgress)
-      }
-    },
-    beforeUnmount: function () {
-      if (window) {
-        window.removeEventListener('beforeunload', this.promptIfInProgress)
-      }
-    },
-    // ...and beforeRouteLeave handles the SPA routing case
-    beforeRouteLeave: function (from, to, next) {
-      // this _shouldn't_ be possible, but seems to be during hot reload in dev
-      if (!this) {
-        return next()
+  definePageMeta({
+    layout: 'no-mobile-cta',
+  })
+
+  const { query } = useRoute()
+  const store = useStore()
+
+  const mode = ref('edit')
+  const previewEvent = ref(null)
+  const partner = ref(null)
+  const submissionErrorMessage = ref(null)
+  const form = ref(null)
```

Comment:
Per [this](https://vuejs.org/guide/essentials/template-refs.html) there's a special helper for template refs:

```suggestion
  const form = useTemplateRef('form'l)
```

Can't say I like the API change here on Vue's part

Reply:
unfortunately useTemplateRef seems to break it, I'm not sure why, but if I switch to that, when I hit the final submit button on the preview screen it just goes into an endless spinner, no errors that I can see, it just gets stuck there :shrug: seems to work fine with ref though

Reply:
```
  const submitEventForReal = () => {
    previewEvent.value = null
    mode.value = 'pending'
    console.log('!!! much upload for real: ' + form.UploadEvent + ', ' + form.value)
    form.value?.UploadEvent()
  }
  ```
  
  if I use useTemplateRef form.value seems to be null, form.UploadEvent also does not exit

Reply:
Interesting, I tried it locally and it worked for both `form` and `previewHeading`. I'm fine with the way you have it, though, seems like this was the pre-Vue 3.5 way (I _really_ don't care for the "magic" inherent in "use the same variable name as the ref in the template")